packages/jsActions/nanoflow-actions-native/src/external/DownloadFileByURL.ts

```diff
@@ -0,0 +1,46 @@
+// This file was generated by Mendix Studio Pro.
+//
+// WARNING: Only the following code will be retained when actions are regenerated:
+// - the import list
+// - the code between BEGIN USER CODE and END USER CODE
+// - the code between BEGIN EXTRA CODE and END EXTRA CODE
+// Other code you write will be lost the next time you deploy the project.
+import fetchBlob from "rn-fetch-blob";
+
+// BEGIN EXTRA CODE
+function formatPath(...pathArgs: string[]): string {
+    return pathArgs.filter(arg => !!arg).join("/");
+}
+
+function validateFileName(name: string) {
+    return name.replace(/[<>"?:|*\/\\\u0000-\u001F\u007F]/g, "_");
+}
+// END EXTRA CODE
+
+/**
+ * @param {string} fileURL - The file Url is required
+ * @param {string} filePath - optional
+ * @param {string} fileName - optional - should be with extension
+ * @returns {Promise.<boolean>}
+ */
+export async function DownloadFileByURL(fileURL: string, filePath: string, fileName: string): Promise<boolean> {
+    // BEGIN USER CODE
+    if (!fileURL) {
+        return Promise.reject(new Error("Input parameter 'file URL' is required"));
+    }
+    const dirs = fetchBlob.fs.dirs;
+    try {
+        const fileNameOverride = fileName || fileURL.substring(fileURL.lastIndexOf("/") + 1);
```

Comment:
I just realized: if this file name is coming from a URL, shouldn't it be URL-decoded? Otherwise you could end up with `%20` or `+` (which is then sanitized to an underscore) instead of an actual space, for example.

Reply:
:sparkles: